mono/utils/mono-threads-api.h

```diff
@@ -22,34 +22,54 @@ MONO_BEGIN_DECLS
 This API is experimental. It will eventually be required to properly use the rest of the raw-omp embedding API.
 */
 
+// FIXME This is duplicated in libtest.c and mono-threads-api.h.
+typedef struct _MonoStackData {
+	const char *function_name;
+} MonoStackData;
+typedef const MonoStackData *MonoStackDataParam; // usual
+typedef       MonoStackData *MonoStackDataParamMutable; // rare
+// const here seems to trigger compiler bug, giving static instead of stack storage
+#define MONO_STACK_DATA(x) /*const*/ MonoStackData  x##_value = { __func__ }; \
+                           const MonoStackDataParam x = &x##_value
+
 MONO_API gpointer
-mono_threads_enter_gc_unsafe_region (gpointer* stackdata);
```

Comment:
All these functions are `MONO_API` and used by the products. Changing them effectively mean you will take care of fixing the products for it.

Reply:
They should still work.

Reply:
I don't think the enter/exit paths are used by the products. I'll check.
Only coop attach/detach still are.

Reply:
oh, my mistake.
jays-MacBook-Pro:xamarin-macios jay$ git grep mono_threads_enter_gc_unsafe_region 
runtime/exports.t4:             new Export (true, "void*", "mono_threads_enter_gc_unsafe_region",
runtime/xamarin/runtime.h:              gpointer __gc_unsafe_cookie = mono_threads_enter_gc_unsafe_region (&__dummy)    \
jays-MacBook-Pro:xamarin-macios jay$ git grep mono_threads_enter_
runtime/exports.t4:             new Export (true, "void*", "mono_threads_enter_gc_unsafe_region",
runtime/exports.t4:             new Export (true, "void*", "mono_threads_enter_gc_safe_region",
runtime/xamarin/runtime.h:              gpointer __gc_unsafe_cookie = mono_threads_enter_gc_unsafe_region (&__dummy)    \
runtime/xamarin/runtime.h:              gpointer __gc_safe_cookie = mono_threads_enter_gc_safe_region (&__dummy)        \
jays-MacBook-Pro:xamarin-macios jay$

Reply:
So I need to treat these like coop attach/detach then and have more variations, or give up on the whole thing..